Sources/System/FilePath/FilePathComponents.swift

```diff
@@ -72,6 +72,16 @@ extension FilePath {
   }
 }
 
+extension FilePath.Component {
```

Comment:
```suggestion
extension FilePath.Component: Hashable {
```

@lorentey would this be possible to do (without availability) as the ABI-stable build picked up the symbols?

Also, would we still need the `public func` below if we added the conformance explicitly?

Reply:
The re-declaration of the conformance is probably fine, ABI-wise: this type already shipped with this.

Reply:
The sinking of the Equatable/Hashable conformances from `_StrSlice` to `Component` is probably fine, ABI-wise. If any other public type conformed to `_StrSlice`, then it'll need to do the same.